src/aspire/source/image.py

```diff
@@ -1309,6 +1309,13 @@ class IndexedSource(ImageSource):
     Map into another into ImageSource.
     """
 
+    _indexed_attrs = [
```

Comment:
This seems awfully specific to a certain application of IndexedSource. Maybe you can elaborate on the use case during our meeting?

At any rate, if we want this, we should think about some way of supplying such *inner* attributes in a more general way (maybe as part of a metadata transfer?).

Reply:
Sure. The use case is that any attribute extending a source which isn't currently implemented (in ASPIRE) as a Relion metadata variable is dropped by IndexedSource.  The list allows devs to add them easily, exposing the slice as the sources proceed.

I thought it was more general than just wrapping the specific methods. We can do these things by metadata table if that's the direction we want to go. Will just need to decide column names 😂 .

Its not totally necessary to do this at all, it was a user request.

Reply:
I implemented this change, but `class_indices` fails because our metadata assignment does not work naturally for 2d fields at this time.  This seems pretty unnatural for the underlying STARFILE format.

Relates to https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/924 , but that is specifically referring to the other kind of 2d case. (multiple fields and multiple values).

I'm looking at whether I can quick fix this case (single field, 2d table) right now and will report back.

Reply:
I've forced our metadata to admit this extra dimension of data. This is going to take a lot of testing for me to feel comfortable with it though ( and its still just another terrible hack). I will try to test it enough that we can do a release before FI, but I'm less sure now.  There are several more serious bugs I should be working on....

Reply:
Added a test to confirm the `metadata` is being written out and read back ( it wasn't, so I guess its for the best that I tested).

Running example recons now, hopefully good to go after that.

Reply:
Yep that's quite the hack. We need to take a long think about how to expose metadata that's being passed down. Maybe some way to override `__getattr__` to see if the attribute exists as a metadata field. Hardcoding all possible metadata fields in `ImageSource` seems like it's going to be a nightmare going forward.